sklearn/preprocessing/imputation.py

```diff
@@ -189,6 +202,15 @@ def _sparse_fit(self, X, strategy, missing_values, axis):
         else:
             X = X.tocsc()
 
+        if axis == 0:
+            mask = _get_mask(X.data, self.missing_values)
+            mask_matrix = X.__class__((mask, X.indices.copy(),
+                                       X.indptr.copy()), shape=X.shape,
+                                      dtype=X.dtype)
+            feat_with_missing = mask_matrix.sum(axis=0).A.nonzero()[1]
```

Comment:
It shouldn't be `.A.nonzero()`. We don't want to transform the sparse matrix to dense.